packages/web-app-files/src/store/actions.ts

```diff
@@ -581,79 +365,98 @@ export default {
    * This will add new entries into the shares tree and will
    * not remove unrelated existing ones.
    */
-  loadSharesTree(context, { client, path, storageId }) {
+  loadSharesTree(context, { client, path, storageId, includeRoot = false }) {
     context.commit('SHARESTREE_ERROR', null)
     // prune shares tree cache for all unrelated paths, keeping only
     // existing relevant parent entries
     context.dispatch('pruneSharesTreeOutsidePath', path)
+    context.commit('INCOMING_SHARES_LOAD', [])
+    context.commit('CURRENT_FILE_OUTGOING_SHARES_SET', [])
+    context.commit('SHARESTREE_LOADING', true)
 
-    const parentPaths = getParentPaths(path, true)
+    const parentPaths = path === '/' && includeRoot ? ['/'] : getParentPaths(path, true)
     const sharesTree = {}
+    const outgoingShares = []
+    const incomingShares = []
 
-    context.commit('SHARESTREE_LOADING', true)
     const shareQueriesQueue = new PQueue({ concurrency: 2 })
     const shareQueriesPromises = []
+    const { highlightedFile } = context.getters
+
+    const getShares = (subPath, indirect, options, outgoing) => {
+      const buildMethod = outgoing ? buildShare : buildCollaboratorShare
+      const resource = indirect || !highlightedFile ? { type: 'folder' } : highlightedFile
+      const arr = outgoing ? outgoingShares : incomingShares
+      const permissions = allowSharePermissions(context.rootGetters)
+      if (!sharesTree[subPath]) {
+        sharesTree[subPath] = []
+      }
+      return client.shares
+        .getShares(subPath, options)
+        .then((data) => {
+          data.forEach((element) => {
+            const share = {
+              ...buildMethod(element.shareInfo, resource, permissions),
+              outgoing,
+              indirect
+            }
+            sharesTree[subPath].push(share)
+            if (!indirect) {
+              arr.push(share)
+            }
+          })
+        })
+        .catch((error) => {
+          console.error('SHARESTREE_ERROR', error)
+          context.commit('SHARESTREE_ERROR', error.message)
+          context.commit('SHARESTREE_LOADING', false)
+          context.commit('CURRENT_FILE_OUTGOING_SHARES_ERROR', error.message)
+          context.commit('INCOMING_SHARES_ERROR', error.message)
+        })
+    }
+
+    if (!path) {
+      // space shares
+      shareQueriesPromises.push(
+        getShares(path, false, { reshares: true, spaceRef: storageId }, true)
+      )
+    }
+
     parentPaths.forEach((queryPath) => {
-      // skip already cached paths
+      const indirect = path !== queryPath
+      // FIXME: We need the storageId of each parent resource here
+      const spaceRef = indirect ? null : storageId
```

Comment:
This is ugly... We can't take the `storageId` of the current resource for each of its parents. This currently results in incorrect indirect shares being displayed when both of the following applies:

a) You are in in nested folder (at least 2 levels deep)
b) You (re-)load the page within the folder. Navigating from the outside into the folder should be fine.